### DISCUSSION TITLE
Mobile server?

### DISCUSSION THREAD

**Discussion by NonzeroCornet** [Thorium-Sim/thorium-nova#298] [Feature Proposals]
Hello. Is Thorium nova going to have a portable mobile server build? I know thorium classic used to have something like that, but the app was taken down. It would just be nice if you need to test something on the fly, you could set up a LAN server.

**Comment by alexanderson1993** [comment]
That's a good question. For starters, the Thorium Classic mobile app was just a supplement; it didn't actually run a server at all. You had to start another server and then connect the mobile app to it.

For me personally, I don't have any desire to create a mobile app. Any mobile functionality for the controls should be done through responsive web design. Maybe we could create a WebView-based app that makes it easier to connect to the main server, but I don't really want to spend too much time on that now.

Hosting a server on a mobile device is an entirely different challenge, since you would have to run Node.js on your mobile devices. Something like this could do the trick: https://github.com/nodejs-mobile/nodejs-mobile. You could take the Node.js build output that normally goes into the Electron app and shove it in the mobile app that's using that framework.

Again, this isn't something that I want to actively pursue, but if someone else in the community wants to do it, I'm happy to support them by putting a link to the app on thoriumsim.com, and even hosting the app on the iOS App Store using my developer account.

**Reply by NonzeroCornet** [reply]
Could I take a crack at it?

**Reply by alexanderson1993** [reply]
Go for it. I've made you a member of the Thorium-Sim org, so you can create a new repo for the mobile app if you want. I would start by making an issue in the Thorium Nova repo explaining what you plan to do, just to make sure we're on the same page. Let me know if you need anything from me.

**Reply by NonzeroCornet** [reply]
Alright cool. Once I'm able I'll start on that.
